resources/geometry.rb

```diff
@@ -19,6 +19,69 @@ def self.make_polygon(*pts)
     return p
   end
 
+  def self.get_thermal_zones(building:)
+    thermal_zones = []
+    building.elements.each("BuildingDetails/Enclosure/Walls/Wall") do |wall|
+      wall_values = HPXML.get_wall_values(wall: wall)
+      thermal_zones << wall_values[:interior_adjacent_to]
+    end
+    building.elements.each("BuildingDetails/Enclosure/FoundationWalls/FoundationWall") do |foundation_wall|
+      foundation_wall_values = HPXML.get_foundation_wall_values(foundation_wall: foundation_wall)
+      thermal_zones << foundation_wall_values[:interior_adjacent_to]
+    end
+    building.elements.each("BuildingDetails/Enclosure/Roofs/Roof") do |roof|
+      roof_values = HPXML.get_roof_values(roof: roof)
+      thermal_zones << roof_values[:interior_adjacent_to]
+    end
+    building.elements.each("BuildingDetails/Enclosure/RimJoists/RimJoist") do |rim_joist|
+      rim_joist_values = HPXML.get_rim_joist_values(rim_joist: rim_joist)
+      thermal_zones << rim_joist_values[:interior_adjacent_to]
+    end
+    building.elements.each("BuildingDetails/Enclosure/FrameFloors/FrameFloor") do |framefloor|
+      framefloor_values = HPXML.get_framefloor_values(framefloor: framefloor)
+      thermal_zones << framefloor_values[:interior_adjacent_to]
+    end
+    building.elements.each("BuildingDetails/Enclosure/Slabs/S;an") do |slab|
+      slab_values = HPXML.get_slab_values(slab: slab)
+      thermal_zones << slab_values[:interior_adjacent_to]
+    end
+    return thermal_zones.uniq
+  end
+  
+  def self.get_model_thermal_zone(model:,
```

Comment:
I don't understand what this method is doing. How is it different than just returning every thermal zone of the model? Why would some zones be excluded?

Reply:
You pass in a string (e.g., "living space"), and the corresponding thermal zone from the model is returned. In this case, we need it since we are manually setting the volume on this model object.

Reply:
Oh, right. 🤦‍♂